File: hw/ramos/u3/hw29Singleton/Java/AlertSystemDay/src/main/java/ec/edu/espe/alertsystemday/model/AlertConfigSingleton.java
package ec.edu.espe.alertsystemday.model;

/**
 *
 * @author Paulo Ramos
 */
import java.io.*;

public class AlertConfigSingleton {

    private static AlertConfigSingleton instance;
    private int alertDays;
    private static final String FILE_PATH = "alert_config.json";

    private AlertConfigSingleton() {
        load();
    }

    public static AlertConfigSingleton getInstance() {
        if (instance == null) {
            instance = new AlertConfigSingleton();
        }
        return instance;
    }

    public int getAlertDays() {
        return alertDays;
    }

    public void setAlertDays(int alertDays) {
        this.alertDays = alertDays;
        save();
    }

    private void load() {
        File file = new File(FILE_PATH);
        if (!file.exists()) {
            alertDays = 3;
            save();
            return;
        }

        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
            alertDays = Integer.parseInt(reader.readLine());
        } catch (IOException e) {
            alertDays = 3;
        }
    }

    private void save() {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH))) {
            writer.write(String.valueOf(alertDays));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: hw/ramos/u3/hw29Singleton/python/AlertSystemPerDay/model/alert_config_singleton.py
class AlertConfigSingleton:
    _instance = None
    _file_path = "alert_config.json"

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(AlertConfigSingleton, cls).__new__(cls)
            cls._instance._alert_days = 3
            cls._instance._load()
        return cls._instance

    @classmethod
    def get_instance(cls):
        return cls()

    def get_alert_days(self) -> int:
        return self._alert_days

    def set_alert_days(self, alert_days: int):
        self._alert_days = alert_days
        self._save()

    def _load(self):
        try:
            with open(self._file_path, "r") as file:
                self._alert_days = int(file.read())
        except (FileNotFoundError, ValueError):
            self._alert_days = 3
            self._save()

    def _save(self):
        try:
            with open(self._file_path, "w") as file:
                file.write(str(self._alert_days))
        except IOError as e:
            print("Error saving configuration:", e)
